src/api/scan/base_controller.go

```diff
@@ -254,39 +297,33 @@ func (bc *basicController) Scan(artifact *v1.Artifact, options ...Option) error
 }
 
 // GetReport ...
-func (bc *basicController) GetReport(artifact *v1.Artifact, mimeTypes []string) ([]*scan.Report, error) {
+func (bc *basicController) GetReport(r *scanner.Registration, artifact *v1.Artifact, mimeTypes []string) ([]*scan.Report, error) {
```

Comment:
If changeing the method signature to this mode, it seems that we should get the scanner meta in the API handler by scanner ctl and then pass it to the scan ctl.

Base on current programming principles, a ctl can call any components under pkg package and provide a full function. That means for the scan ctl, it can still provide scan (art, options) method.

Reply:
Get registration of the project depend on scanner manager and project metadata manager,  so the `GetRegistrationByProject` can't be implemented in scanner manager.

If we process GetRegistrationByProject  in scan controller , it will  be duplicated with scanner controller.